src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -30,7 +31,18 @@ public EndpointParameter(IParameterSymbol parameter, WellKnownTypes wellKnownTyp
         {
             Source = EndpointParameterSource.Query;
             AssigningCode = $"httpContext.Request.Query[\"{parameter.Name}\"]";
-            IsOptional = parameter.Type is INamedTypeSymbol parameterType && parameterType.NullableAnnotation == NullableAnnotation.Annotated;
+            IsOptional = parameter.Type is INamedTypeSymbol
+            {
+                NullableAnnotation: NullableAnnotation.Annotated
+            };
+        }
+        else if (parameter.HasAttributeImplementingInterface(fromServiceMetadataInterfaceType))
```

Comment:
Can we check these in the same relative order as RDF? It shouldn't have a functional impact yet, but it should make comparison easier.

So, it'd be:

1. Attributes (route, query, header, body, form, services, as-parameters, in that order)
2. "Well known" types
3. BindAsync
4. TryParse (and string)
5. Arrays and StringValues
6. Service or body

Reply:
Yep, that's probably sensible. I'll update the ordering for the things we currently support.